cli/src/cmd/forge/script/broadcast.rs

```diff
@@ -0,0 +1,205 @@
+use crate::{
+    cmd::{
+        forge::script::receipts::{get_pending_txes, maybe_has_receipt, wait_for_receipts},
+        ScriptSequence,
+    },
+    utils::{get_http_provider, print_receipt},
+};
+use ethers::{
+    prelude::{k256::ecdsa::SigningKey, Http, Provider, SignerMiddleware, TxHash, Wallet},
+    providers::Middleware,
+    types::{transaction::eip2718::TypedTransaction, Chain, TransactionReceipt},
+};
+
+use super::*;
+
+impl ScriptArgs {
+    pub async fn send_transactions(
+        &self,
+        deployment_sequence: &mut ScriptSequence,
+        fork_url: &str,
+    ) -> eyre::Result<()> {
+        let provider = get_http_provider(fork_url);
+        let chain = provider.get_chainid().await?.as_u64();
+
+        let required_addresses = deployment_sequence
+            .transactions
+            .iter()
+            .map(|tx| *tx.from().expect("No sender for onchain transaction!"))
+            .collect();
+
+        let local_wallets = self.wallets.find_all(chain, required_addresses)?;
+        if local_wallets.is_empty() {
+            eyre::bail!("Error accessing local wallet when trying to send onchain transaction, did you set a private key, mnemonic or keystore?")
+        }
```

Comment:
Let's follow up with Ledger/Trezor support here @joshieDo in a separate PR?

Reply:
https://github.com/foundry-rs/foundry/pull/1764